test/e2e/lib/helpers/recorder.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  NodeType,
+  SerializedNode,
+  DocumentNode,
+  ElementNode,
+  TextNode,
+} from '../../../../packages/rum-recorder/src/domain/rrweb-snapshot/types'
+import {
+  Segment,
+  RecordType,
+  FullSnapshotRecord,
+  MetaRecord,
+  IncrementalSnapshotRecord,
+  IncrementalSource,
+} from '../../../../packages/rum-recorder/src/types'
+
+// Returns this first MetaRecord in a Segment, if any.
+export function findMeta(segment: Segment): MetaRecord | null {
+  return segment.records.find((record) => record.type === RecordType.Meta) as MetaRecord
+}
+
+// Returns this first FullSnapshotRecord in a Segment, if any.
+export function findFullSnapshot(segment: Segment): FullSnapshotRecord | null {
+  return segment.records.find((record) => record.type === RecordType.FullSnapshot) as FullSnapshotRecord
+}
+
+// Returns the first IncrementalSnapshotRecord of a given source in a
+// Segment, if any.
+export function findIncrementalSnapshot(segment: Segment, source: IncrementalSource): IncrementalSnapshotRecord | null {
+  return segment.records.find(
+    (record) => record.type === RecordType.IncrementalSnapshot && record.data.source === source
+  ) as IncrementalSnapshotRecord
+}
```

Comment:
Here, the typing *could* be improved with type guards too, for example:

```ts
const allInputData = segment.records.find(isIncrementalSnapshot).map(record => record.data).filter(isInputData)
```

But it would be a bit more verbose, I'll let you decide if it's worth it :p

Reply:
i see your point but it feels quite verbose adding these isXX functions here as well. maybe we add them only if/when other functions start requiring it?